graphs/util/db.py

```diff
@@ -1608,6 +1608,298 @@ def find_all_graphs_containing_nodes(uid, search_type, search_word, view_type, d
+			if 'TextLabel' in node:
+				tempNode['data']['content'] = node['TextLabel']
+
+			# Deal with the ShapeType field of the shape element,
+			# improve this, make it more general, too many hacks right now.
+			if 'ShapeType' in node:
+				if node['ShapeType'].lower() == 'brace':
+					tempNode['data']['shape'] = 'rectangle'
+				else:
+					# oval in GPML is ellipse in cytoscape JSON
+					if node['ShapeType'].lower() == 'oval':
+						tempNode['data']['shape'] = 'ellipse'
+					# roundedrectangle in GPML is roundrectangle in cytoscape JSON
+					elif node['ShapeType'].lower() == 'roundedrectangle':
+						tempNode['data']['shape'] = 'roundrectangle'
+					# everything else works right now, may break with more testing
+					else:
+						tempNode['data']['shape'] = node['ShapeType'].lower()
+			else:
+				# if no ShapeType given use rectangle by default.
+				tempNode['data']['shape'] = 'rectangle'
+
+			if not ('Height' in node and 'Width' in node):
+				return {"Error": "GPML file must contain the height and width of all the shape elements!"}
+			tempNode['data']['height'] = node['Height']
+			tempNode['data']['width'] = node['Width']
+			# this is the twist out here, if you compare this with node color attribute
+			# we use the color given for border_color rather than color (i.e. text color)
+			if 'Color' in node:
+				tempNode['data']['border_color'] = '#' + node['Color']
+			if 'FillColor' in node:
+				tempNode['data']['background_color'] = '#' + node['FillColor']
+			else:
+				tempNode['data']['background_color'] = 'white'
+			# this has to be zero because we'll have multiple elements
+			tempNode['data']['background_opacity'] = 0
+			if 'key' in node:
+				if node['key'] == 'org.pathvisio.DoubleLineProperty':
+					tempNode['data']['border_style'] = 'double'
+			if 'LineThickness' in node:
+				tempNode['data']['border_width'] = node['LineThickness']
+			else:
+				tempNode['data']['border_width'] = 1
+			# increase count, i.e. to keep unique ids for shapes, nodes, labels.
+			count = count + 1
+			parseJson['graph']['nodes'].append(tempNode)
+
+
+		# Find all the DataNodes in the GPML file which is used to draw various
+		# the various nodes of the graph
+		datanode = tree.findall(GPML+'DataNode')
+
+		# Parsing the DataNode element for all the details.
+		node_details = []
+		for node in datanode:
+		    temp = {}
+		    temp = node.attrib
+		    for details in node:
+		        temp.update(details.attrib)
+		    node_details.append(temp)
+
+		# Go through the node details of the node
+		for node in node_details:
+			tempNode = {'data': {}}
+
+			# Give an id to the node element.
+			tempNode['data']['id'] = count
+			if not ('CenterX' in node and 'CenterY' in node):
+				return {"Error": "GPML file must contain X and Y coordinates of all the DataNode elements!"}
+			tempNode['data']['x'] = node['CenterX']
+			tempNode['data']['y'] = node['CenterY']
+			# Not sure about text label. Haven't got an error yet for textlable, YET!!
```

Comment:
Can describe your concerns here ? Also make it a TODO item if you think we need to review/test it later.